homeassistant/components/ultrasync/coordinator.py

```diff
@@ -0,0 +1,105 @@
+        """Initialize global UltraSync data updater."""
+        self.hub = ultrasync.UltraSync(
+            user=config[CONF_USERNAME],
+            pin=config[CONF_PIN],
+            host=config[CONF_HOST],
+        )
+
+        # Used to track delta (for change tracking)
+        self._area_delta = {}
+        self._zone_delta = {}
+
+        update_interval = timedelta(seconds=options[CONF_SCAN_INTERVAL])
+
+        super().__init__(
+            hass,
+            _LOGGER,
+            name=DOMAIN,
+            update_interval=update_interval,
+        )
+
+    async def _async_update_data(self) -> dict:
+        """Fetch data from UltraSync Hub."""
+
+        def _update_data() -> dict:
+            """Fetch data from UltraSync via sync functions."""
+
+            # initialize our response
+            response = {}
+
+            # Update our details
+            details = self.hub.details(max_age_sec=0)
+            if details:
+                async_dispatcher_send(
+                    self.hass,
+                    SENSOR_UPDATE_LISTENER,
+                    details["areas"],
+                    details["zones"],
+                )
+
+                for zone in details["zones"]:
+                    if self._zone_delta.get(zone["bank"]) != zone["sequence"]:
+                        self.hass.bus.fire(
+                            "ultrasync_zone_update",
+                            {
+                                "sensor": zone["bank"] + 1,
+                                "name": zone["name"],
+                                "status": zone["status"],
+                            },
+                        )
+
+                        # Update our sequence
+                        self._zone_delta[zone["bank"]] = zone["sequence"]
+
+                    # Set our state:
+                    response["zone{:0>2}_state".format(zone["bank"] + 1)] = zone[
+                        "status"
+                    ]
+
+                for area in details["areas"]:
+                    if self._area_delta.get(area["bank"]) != area["sequence"]:
+                        self.hass.bus.fire(
+                            "ultrasync_area_update",
+                            {
+                                "area": area["bank"] + 1,
+                                "name": area["name"],
+                                "status": area["status"],
+                            },
+                        )
+
+                        # Update our sequence
+                        self._area_delta[area["bank"]] = area["sequence"]
+
+                    # Set our state:
+                    response["area{:0>2}_state".format(area["bank"] + 1)] = area[
+                        "status"
+                    ]
+
+            # Return our response
+            return response
+
+        # The hub can sometimes take a very long time to respond; wait
+        # 10 seconds before giving up
+        async with timeout(10):
```

Comment:
This doesn't make sense. We have a poll interval of 1 second and a timeout of 10 seconds? The poll interval must be longer than the maximum time it takes to complete the poll.

Reply:
The issue here is 9/10 the poll will return in under 1 second. So this is just more of a cautionary handling of the 1 off (so it breaks out) (let setting a SIGALRM if you will).

Reply:
If it can happen 1/10 my comment above still needs to be resolved.